File: frontend/src/app/models/work-order.ts
export interface WorkOrder {
  id: number;
  title: string;
  description: string;
  status: 'Open' | 'Assigned' | 'In Progress' | 'Waiting on Parts' | 'Completed' | 'Closed';
  priority: 'Low' | 'Medium' | 'High' | 'Urgent';
  assignedTo: string;
  createdBy: string;
  createdAt: string;
  dueDate: string;
}

File: backend/Models/WorkOrder.cs
namespace backend.Models;

public class WorkOrder
{
    public int Id { get; set; }

    public string Title { get; set; } = "";
    public string Description { get; set; } = "";
    public string Status { get; set; } = "";
    public string Priority { get; set; } = "";
    public string AssignedTo { get; set; } = "";
    public string CreatedBy { get; set; } = "";

    public DateTime CreatedAt { get; set; }
    public DateTime DueDate { get; set; }

    public List<WorkOrderActivity> Activities { get; set; } = new();
}
